contracts/common-utils/src/math.rs

```
@@ -225,3 +225,136 @@ pub fn compound_interest(
+pub fn median_of_n(values: &[i128]) -> core::result::Result<i128, CommonError> {
+    if values.is_empty() {
+        return Err(CommonError::InvalidInput);
+    }
+
+    let mut sorted = values.to_vec();
+    sorted.sort();
+
+    let len = sorted.len();
+    let mid = len / 2;
+
+    if len % 2 == 0 {
```

Clippy (1.97.1)

[warning] manual implementation of `.is_multiple_of()`

(warning)